samples/getting_started/SimpleTeleportation.qs

```diff
@@ -0,0 +1,51 @@
+/// # Summary
+/// Simple quantum teleportation sample
+///
+/// # Description
+/// This Q# program demonstrates how to teleport quantum state
+/// by communicating two classical bits using previously entangled qubits.
+/// This code teleports |1〉 state, but any state can be teleported.
+operation Main() : Bool {
+    // Allocate qAlice, qBob qubits
+    use (qAlice, qBob) = (Qubit(), Qubit());
+
+    // Entangle qAlice, qBob qubits
+    H(qAlice);
+    CNOT(qAlice, qBob);
+
+    // Allocate qMessage qubit and prepare it to be |1〉
+    use qMessage = Qubit();
+    X(qMessage);
+
+    // Prepare the message by entangling with qAlice state
+    CNOT(qMessage, qAlice);
+    H(qMessage);
+
+    // Obtain classical measurement results b1 and b2
+    let b1 = M(qMessage) == One;
+    let b2 = M(qAlice) == One;
+
+    // Here classical results b1 and b2 are "sent" to the Bob's site.
+
+    // Decode the message by applying adjustments based on classical data b1 and b2.
+    if b1 {
+        Z(qBob);
+    }
+    if b2 {
+        X(qBob);
+    }
+
+    // Make sure that the obtained result is the same
+    X(qBob);
+    let correct = Std.Diagnostics.CheckZero(qBob);
```

Comment:
Hmm... using `CheckZero` here makes the verification deterministically correct, but I worry it might be more confusing than just using measurement. Is there a reason to not use measurement here without the `X` and check that the expected `One` is returned?

Reply:
The reason is the determinism you mentioned. Just one execution is enough to make sure that the resulting state is correct. Also, if I use measurements, I need to change the message. I can no longer assert that the **state** is correct, I can only say that the **measurement result** is correct (and state may still be incorrect).
Also, I'm not impressed by teleporting |1> state. Feels like communicating two classical bits defeats the trick in this case. A more complex state would be more interesting to teleport...